src/account.ts

```diff
@@ -0,0 +1,231 @@
+            BigInt("0x" + signedMessage[1].toString(16))
+        ];
+        return signature;
+    }
+}
+
+/**
+ * Wrapper for the OpenZeppelin implementation of an Account
+ */
+export class OpenZeppelinAccount extends Account {
+    constructor(
+        starknetContract: StarknetContract,
+        privateKey: string,
+        publicKey: string,
+        keyPair: any
+    ) {
+        super(starknetContract, privateKey, publicKey, keyPair);
+    }
+
+    async invoke(
+        toContract: StarknetContract,
+        functionName: string,
+        calldata?: StringMap
+    ): Promise<void> {
+        const { args, signature } = await this.adaptArgs(
+            toContract.address,
+            functionName,
+            calldata
+        );
+        await this.starknetContract.invoke("execute", args, { signature: signature });
+    }
+
+    async call(
+        toContract: StarknetContract,
+        functionName: string,
+        calldata?: StringMap
+    ): Promise<StringMap> {
+        const toAddress = toContract.address;
+        const abi = toContract.getAbi();
+        const { args, signature } = await this.adaptArgs(toAddress, functionName, calldata);
+        const { response: result } = await this.starknetContract.call("execute", args, {
+            signature
+        });
+        const func = <starknet.CairoFunction>abi[functionName];
+        return adaptOutput(result.join(" "), func.outputs, abi);
+    }
+
+    async adaptArgs(toAddress: string, functionName: string, calldata: StringMap) {
+        const { res: nonce } = await this.starknetContract.call("get_nonce");
+        const functionSelector = hash.starknetKeccak(functionName).toString();
+        const calldataArray = calldata ? this.calldataToNumeric(calldata, []) : [];
+        const signature = this.sign(nonce.toString(), functionSelector, toAddress, calldataArray);
+        const args = {
+            to: BigInt(toAddress),
+            selector: functionSelector,
+            calldata: calldataArray.map((it) => BigInt(it.toString())),
+            nonce: nonce.toString()
+        };
+
+        return { args, signature };
+    }
+
+    static async deployFromABI(
+        accountContract: string,
+        hre: HardhatRuntimeEnvironment
+    ): Promise<Account> {
+        const privateKey = generateRandomStarkPrivateKey();
+        const keyPair = getKeyPair(privateKey);
+        const publicKey = getStarkKey(keyPair);
+
+        const contractFactory = await hre.starknet.getContractFactory(accountContract);
+        const contract = await contractFactory.deploy({ _public_key: BigInt(publicKey) });
+
+        const pk = "0x" + privateKey.toString(16);
+        console.log("Account private key: " + pk);
+        console.log("Account public key: " + publicKey);
+        console.log("Account address: " + contract.address);
+
+        return new OpenZeppelinAccount(contract, pk, publicKey, keyPair);
+    }
+
+    static async getAccountFromAddress(
```

Comment:
Instead of having static functions for each Account implementation and starknet utility functions calling them, we could do this:
- The constructor could take parameters, potentially even an options object, and generate/retrieve what it doesn't have.
- We wouldn't need the switch-case logic which would require future updates with new Account implementations.

Reply:
Dropping this idea, no dev-friendly and user-friendly way of implementing it.